starfish/core/_display.py

```diff
@@ -190,49 +194,52 @@ def display(
     Notes
     -----
     - To use in ipython, use the `%gui qt5` magic.
-    - Napari axes currently cannot be labeled. Until such a time that they can, this function will
+    - napari axes currently cannot be labeled. Until such a time that they can, this function will
       order them by Round, Channel, and Z.
-    - Requires napari 0.0.6: install starfish using `pip install starfish[napari]` to install all
+    - Requires napari 0.0.9: install starfish using `pip install starfish[napari]` to install all
       necessary requirements
     """
     if stack is None and spots is None:
+        # masks without stack or spots have no context so don't check for that
         raise TypeError("expected a stack and/or spots; got nothing")
 
-    if Window is None or Viewer is None:
-        warnings.warn("Requires napari 0.0.6. Run `pip install starfish[napari]` to install the "
-                      "necessary requirements.")
-        return
-
-    from PyQt5.QtWidgets import QApplication
+    try:
+        import napari
+        if napari.__version__ != NAPARI_VERSION:
+            raise Exception
```

Comment:
I would prefer that this be a specific exception type, or an exception with a specific payload that can be inspected to insure that it's what we expect.